src/main/java/com/carpool/car_pool/repositories/RideRequestRepository.java

```diff
@@ -1,13 +1,16 @@
 package com.carpool.car_pool.repositories;
 
 import com.carpool.car_pool.repositories.entities.RideRequestsEntity;
+import com.carpool.car_pool.repositories.entities.UserEntity;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
+import java.util.List;
 
 import java.util.Optional;
 
 @Repository
 public interface RideRequestRepository extends JpaRepository<RideRequestsEntity, Long> {
     Optional<RideRequestsEntity> findByRideOfferId(Long rideOfferId);
+    List<RideRequestsEntity> findByRequester(UserEntity requester);
```

Comment:
Has to be Optional since user can have 0 (null) ride requests. Otherwise we can lend in exception that fails the application

Reply:
But if there are more than one ride requests that the user have made, it fails if it is optional

Reply:
If the user has 0 ride requests the method returns an empty list, so no errors